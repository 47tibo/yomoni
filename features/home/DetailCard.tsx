import { Box } from "@/components/ui/box";
import { Heading } from "@/components/ui/heading";
import { Image } from "@/components/ui/image";
import { LinearGradient } from "@/components/ui/linear-gradient";
import { Pressable } from "@/components/ui/pressable";
import { Text } from "@/components/ui/text";
import { VStack } from "@/components/ui/vstack";
import { fromCSS } from "@bacons/css-to-expo-linear-gradient";
import { Link } from "expo-router";

export type DetailCardProps = {
  imageUrl: string;
  title: string;
  subtitle: string;
  objectNumber: string;
};

export default function DetailCard({
  imageUrl,
  title,
  subtitle,
  objectNumber,
}: DetailCardProps) {
  return (
    <Link href={`/art-object/${objectNumber}`} asChild>
      <Pressable>
        <LinearGradient
          className="rounded-md"
          {...fromCSS(`linear-gradient(135deg, #ffffff, #a8a8a8);`)}
        >
          <Box className="w-full h-64 items-center justify-center">
            <Image
              size="2xl"
              height={100}
              width={100}
              source={imageUrl}
              alt={title}
            />
          </Box>
          <Box className="h-4" />
          <VStack space="md" className="p-4">
            <Heading size="md">{title}</Heading>
            <Text className="text-sm">{subtitle}</Text>
          </VStack>
          <Box className="h-4" />
        </LinearGradient>
      </Pressable>
    </Link>
  );
}
