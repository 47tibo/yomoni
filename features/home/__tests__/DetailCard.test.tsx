import { render } from "@testing-library/react-native";
import DetailCard from "../DetailCard";

test("DetailCard matches snapshot", () => {
  const { toJSON } = render(
    <DetailCard imageUrl={""} title={""} subtitle={""} objectNumber={""} />
  );
  expect(toJSON()).toMatchSnapshot();
});
