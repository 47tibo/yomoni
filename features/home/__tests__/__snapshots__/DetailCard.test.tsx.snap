// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailCard matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  className="data-[focus-visible=true]:outline-none data-[focus-visible=true]:ring-indicator-info data-[focus-visible=true]:ring-2 data-[disabled=true]:opacity-40"
  collapsable={false}
  dataSet={
    {
      "active": "false",
      "disabled": "false",
      "focus": "false",
      "focusVisible": "false",
      "hover": "false",
    }
  }
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  states={
    {
      "active": false,
      "disabled": undefined,
      "focus": false,
      "focusVisible": false,
      "hover": false,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        4289243304,
        4294967295,
      ]
    }
    endPoint={
      [
        0,
        1.1102230246251565e-16,
      ]
    }
    locations={
      [
        0,
        1,
      ]
    }
    startPoint={
      [
        1,
        0.9999999999999999,
      ]
    }
  >
    <View
      className="w-full h-64 items-center justify-center"
    >
      <Image
        alt=""
        aria-label=""
        className="max-w-full h-64 w-64"
        height={100}
        role="img"
        source={
          {
            "uri": "",
          }
        }
        width={100}
      />
    </View>
    <View
      className="h-4"
    />
    <View
      className="flex-col gap-3 p-4"
    >
      <Text
        accessibilityRole="header"
        style={
          {
            "fontSize": 26.56,
            "fontWeight": "bold",
            "marginVertical": 53.44,
          }
        }
      />
      <Text
        className="text-typography-700 font-body text-sm"
      />
    </View>
    <View
      className="h-4"
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
